syntax = 'proto3';

package authentication;


service Authentication {
    rpc Login(LoginRequest) returns (Response);
    rpc Signup(SignupRequest) returns (Response);
    rpc Query(QueryRequest) returns (QueryResponse);
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message SignupRequest {
    string email = 1;
    string username = 2;
    string password = 3;
}

message Response {
    string token = 1;
    float time = 2;
    int32 status = 3;
}

message QueryRequest {
    string api = 1;
}

message QueryResponse {
    string res = 1;
    float time = 2;
    int32 status = 3;
}
